src/signals/shared/type-guards/index.ts

```diff
@@ -0,0 +1,10 @@
+import type { FetchError } from 'hooks/useFetch'
+
+export const isFetchError = (
+  error?: boolean | FetchError
+): error is FetchError => {
```

Comment:
whats happing here?

Reply:
It's a type guard:) In some cases the error can only be a Fetch error, this make sure only Fetch errors pass. Else you get a lot of typing issues. 
https://www.typescriptlang.org/docs/handbook/advanced-types.html